jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/KModuleDeploymentService.java

```diff
@@ -222,12 +223,12 @@ protected RegisterableItemsFactory getRegisterableItemsFactory(AuditEventBuilder
     }
 
     @Override
-	public void undeploy(DeploymentUnit unit) {
+	public void undeploy(DeploymentUnit unit, Function<DeploymentUnit, Boolean> beforeUndeploy) {
```

Comment:
I'd recommend using `Predicate<DeploymentUnit>` instead of `Function<DeploymentUnit, Boolean>` for this. Apart from being shorter you don't have to deal with the case when user potentially returns null (which is valid Boolean value)

Reply:
there is no need to deal with null as only true is the important one ... everything else will produce an error

Reply:
I think if you return Boolean as null, then if (null) will fail with NPE in the undeploy method...

Reply:
that's true but we are not doing defencing programming to guard any possible user mistakes, if someone returns null then it's expected to fail with NPE